api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -705,11 +710,26 @@ private void updateProfileForUser(DbUser user, Profile updatedProfile, Profile p
     user.setLastModifiedTime(now);
 
     updateInstitutionalAffiliations(updatedProfile, user);
-    if (workbenchConfigProvider.get().featureFlags.requireInstitutionalVerification) {
+    boolean requireInstitutionalVerification =
+        workbenchConfigProvider.get().featureFlags.requireInstitutionalVerification;
+    if (requireInstitutionalVerification) {
       profileService.validateInstitutionalAffiliation(updatedProfile);
     }
 
     userService.updateUserWithConflictHandling(user);
+    if (requireInstitutionalVerification) {
```

Comment:
this whole block is kind of janky and I expect the entire thing to get deleted by RW-4980.